### PR TITLE
Add scenerio to disable/enable viostor/vioscsi driver

### PR DESCRIPTION
**Pull request by suqinhuang**
Signed-off-by: Suqin Huang <shuang@redhat.com>

id: 1231076 1231068

### CODE REVIEW

**Comment by suqinhuang** [comment]
base on https://github.com/autotest/tp-qemu/pull/517

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 1]
```diff
@@ -0,0 +1,50 @@
+- driver_load_stress:
+    type = driver_in_use
+    start_vm = yes
+    kill_vm_on_error = yes
+    check_guest_bsod = yes
+    kill_vm_on_error = yes
+    login_timeout = 240
+    suppress_exception = False
+    run_bg_flag = "before_bg_test"
+    sleep_time = 300
```
Comment (spcui) [inline qemu/tests/cfg/driver_load_stress.cfg]: why need sleep 300 seconds?

Comment (suqinhuang) [inline qemu/tests/cfg/driver_load_stress.cfg]: request by the case

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 2]
```diff
@@ -0,0 +1,50 @@
+- driver_load_stress:
+    type = driver_in_use
```
Comment (spcui) [inline qemu/tests/cfg/driver_load_stress.cfg]: not found driver_in_use.py

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    sleep_time = 300
+    driver_load_cmd = C:\devcon.exe enable @DRIVER_ID
+    driver_unload_cmd = C:\devcon.exe disable @DRIVER_ID
+    driver_id_cmd = C:\devcon.exe find * | find "VirtIO"
+    only Windows
+    variants:
+        - with_block:
+            sub_test = driver_load
+            driver_name = viostor
+            run_bg_stress = iozone_windows
```
Comment (spcui) [inline qemu/tests/cfg/driver_load_stress.cfg]: you can move these lines above variants, cause these are public config.

Comment (suqinhuang) [inline qemu/tests/cfg/driver_load_stress.cfg]: Hi scui,
netdev, baloon, rng, serial drivers will be tested as well.

Regards,

File: qemu/tests/cfg/driver_load_stress.cfg (hunk 2 above)
Comment (suqinhuang) [inline qemu/tests/cfg/driver_load_stress.cfg]: did you use latest code ?

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 4]
```diff
@@ -0,0 +1,50 @@
+- driver_load_stress:
+    type = driver_in_use
+    start_vm = yes
+    kill_vm_on_error = yes
+    check_guest_bsod = yes
+    kill_vm_on_error = yes
+    login_timeout = 240
+    suppress_exception = False
+    run_bg_flag = "before_bg_test"
```
Comment (CongSmile) [inline qemu/tests/cfg/driver_load_stress.cfg]: @suqinhuang 
Could you explain why it's "before_bg_test" only?

Thanks.

**Comment by CongSmile** [comment]
Hi @suqinhuang 

Could you add a flag to control when run the subtest during driver in use?

refer: env[netperf_run] in netperf_stress.py

Thanks.

**Comment by suqinhuang** [comment]
@CongSmile 
As we discussed, I didn't set flag to turn on/off iozone_windows, iozone test could not be stop until it finished, so I set the wait_bg_time to 300s.

Could you pls review it again.

Regards,
Suqin

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 5]
```diff
@@ -0,0 +1,54 @@
+- driver_load_stress:
+    type = driver_in_use
+    start_vm = yes
+    kill_vm_on_error = yes
+    check_guest_bsod = yes
+    kill_vm_on_error = yes
+    login_timeout = 240
+    suppress_exception = False
+    sleep_time = 300
```
Comment (CongSmile) [inline qemu/tests/cfg/driver_load_stress.cfg]: @suqinhuang 
Could you help explain why so long for the sleep time? I have thought the cmd execution is hang.
sleep_time = 300 means the case needs 10mins at least and it's just for the one round.

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            run_bgstress = iozone_windows
+            cd_format_fixed = ide
+            drive_format_image1 = ide
+            cdrom_cd1 = isos/windows/winutils.iso
+            images += " stg"
+            image_name_stg = "images/storage"
+            image_size_stg = 4G
+            drive_format_stg = scsi-hd
+            force_create_image_stg = yes
+            iozone_cmd = "%s:\Iozone\iozone.exe -azR -r 64k -n 1G -g 4G -M -b iozone.xls -f %s:\testfile"
```
Comment (CongSmile) [inline qemu/tests/cfg/driver_load_stress.cfg]: Where give the drive letter?

       Error using popen() on uname
        -M option suppressed.
        Command line used: /Iozone/iozone -azR -r 64k -n 1G -g 4G -M -b iozone.xls -f :\testfile
        Output is in Kbytes/sec
        Time Resolution = 0.000012 seconds.
        Processor cache size set to 1024 Kbytes.
        Processor cache line size set to 32 bytes.
        File stride size set to 17 * record size.
                                                            random  random    bkwd   record   stride    
              KB  reclen   write rewrite    read    reread    read   write    read  rewrite     read   fwrite frewrite   fread  freread
         1048576      64cygwin warning:
  MS-DOS style path detected: :\testfile
  Preferred POSIX equivalent is: :/testfile
  CYGWIN environment variable option "nodosfilewarning" turns off this warning.
  Consult the user's guide for more details about POSIX paths:
    http://cygwin.com/cygwin-ug-net/using.html#using-pathnames

Can not open temp file: :\testfile
open: No such file or directory

File: qemu/tests/cfg/driver_load_stress.cfg (hunk 5 above)
Comment (suqinhuang) [inline qemu/tests/cfg/driver_load_stress.cfg]: 300s is requested in the case

File: qemu/tests/driver_load.py [hunk 7]
```diff
@@ -85,11 +85,11 @@ def check_driver(session, cmd, pattern):
 
         error.context("Unload the driver", logging.info)
         unload_driver(session, driver_unload_cmd, driver_id)
-        time.sleep(5)
+        time.sleep(sleep_time)
 
         error.context("Load the driver", logging.info)
         load_driver(session, driver_load_cmd, driver_id)
-        time.sleep(5)
+        time.sleep(sleep_time)
```
Comment (CongSmile) [inline qemu/tests/driver_load.py]: Confirmed with the feature owner, no need sleep a while after load driver.

**Comment by suqinhuang** [comment]
base on https://github.com/autotest/tp-qemu/pull/546

**Comment by suqinhuang** [comment]
@spcui @CongSmile could you review this patch if you have time.

Thanks a lot

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            sub_test = driver_load
+            driver_name = viostor
+            run_bgstress = iozone_windows
+            drive_format_image1 = ide
+            cdrom_cd1 = isos/windows/winutils.iso
+            images += " stg"
+            image_name_stg = "images/storage"
+            image_size_stg = 4G
+            drive_format_stg = virtio
+            force_create_image_stg = yes
```
Comment (spcui) [inline qemu/tests/cfg/driver_load_stress.cfg]: add remove_image_stg = yes is better.

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    only Windows
+    variants:
+        - before_bg_test:
+            run_bg_flag = "before_bg_test"
+        - during_bg_test:
+            run_bg_flag = "during_bg_test"
+        - after_bg_test:
+            run_bg_flag = "after_bg_test"
+    variants:
+        - with_block:
```
Comment (CongSmile) [inline qemu/tests/cfg/driver_load_stress.cfg]: Need set flag "suppress_exception" to handle the exception when run subtest.

Comment (suqinhuang) [inline qemu/tests/cfg/driver_load_stress.cfg]: line 8: +    suppress_exception = False

**Comment by suqinhuang** [comment]
@spcui updated, could you review again

**Comment by spcui** [comment]
LGTM
Acked-by: scui@redhat.com

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+            image_name_stg = "images/storage"
+            image_size_stg = 4G
+            drive_format_stg = virtio
+            force_create_image_stg = yes
+            remove_image_stg = yes
+            iozone_cmd = "%s:\Iozone\iozone.exe -azR -r 64k -n 1G -g 4G -M -b iozone.xls -f %s:\testfile"
+            iozone_timeout = 7200
+            wait_bg_time = 180
+            driver_id_pattern = "(.*?):.*?VirtIO SCSI Disk Device"
+            disk_letter = I
```
Comment (CongSmile) [inline qemu/tests/cfg/driver_load_stress.cfg]: For the disk_letter is fixed, is there another way for the problem?
It's a little wired.

Comment (suqinhuang) [inline qemu/tests/cfg/driver_load_stress.cfg]: need to provide a disk_letter or use default one when create a partition

**Comment by CongSmile** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Comment by suqinhuang** [comment]
Thanks @spcui @CongSmile merged

**Pull request merged**